packages/runtime/src/api/react.tsx

```diff
@@ -1,157 +1,152 @@
+} from './graphql/generated/types'
 
-const typePolicies: TypePolicies = {
-  Query: {
-    fields: {
-      swatches(existingData, { args, toReference }) {
-        return (
-          existingData ??
-          args?.ids.map((id: string) => toReference({ __typename: 'Swatch', id }, true))
-        )
-      },
-      file(existingData, { args, toReference }) {
-        return existingData ?? toReference({ __typename: 'File', id: args?.id }, true)
-      },
-      files(existingData, { args, toReference }) {
-        return (
-          existingData ??
-          args?.ids.map((id: string) => toReference({ __typename: 'File', id }, true))
-        )
-      },
-      typographies(existingData, { args, toReference }) {
-        return (
-          existingData ??
-          args?.ids.map((id: string) => toReference({ __typename: 'Typography', id }, true))
-        )
-      },
-      pagePathnamesById(existingData, { args, toReference }) {
-        return (
-          existingData ??
-          args?.ids.map((id: string) => toReference({ __typename: 'PagePathnameSlice', id }, true))
-        )
-      },
-      globalElement(existingData, { args, toReference }) {
-        return existingData ?? toReference({ __typename: 'GlobalElement', id: args?.id }, true)
-      },
-      table(existingData, { args, toReference }) {
-        return existingData ?? toReference({ __typename: 'Table', id: args?.id }, true)
-      },
-      page(existingData, { args, toReference }) {
-        return existingData ?? toReference({ __typename: 'Page', id: args?.id }, true)
-      },
-      site(existingData, { args, toReference }) {
-        return existingData ?? toReference({ __typename: 'Site', id: args?.id }, true)
-      },
-    },
-  },
+export type MakeswiftClientOptions = {
+  uri: string
+  serializedState?: MakeswiftApiClient.SerializedState
 }
 
-type CreateApolloClientParams = {
-  uri?: string
-  cacheData?: NormalizedCacheObject
-}
+// NOTE(miguel): MakeswiftClient -> HostClient
+export class MakeswiftClient {
+  graphqlClient: GraphQLClient
+  makeswiftApiClient: MakeswiftApiClient.Store
+  subscribe: MakeswiftApiClient.Store['subscribe']
+
+  constructor({ uri, serializedState }: MakeswiftClientOptions) {
+    this.graphqlClient = new GraphQLClient(uri)
+    this.makeswiftApiClient = MakeswiftApiClient.configureStore({
+      graphqlClient: this.graphqlClient,
+      serializedState,
+    })
+    this.subscribe = this.makeswiftApiClient.subscribe
+  }
 
-export function createApolloClient({ uri, cacheData }: CreateApolloClientParams) {
-  const cache = new InMemoryCache({ typePolicies })
+  async prefetch(element: Element): Promise<{
+    serializedState: MakeswiftApiClient.SerializedState
+  }> {
+    const introspectionData = await introspect(element, this, storeContextDefaultValue)
 
-  if (cacheData) cache.restore(cacheData)
+    await this.makeswiftApiClient.dispatch(
+      MakeswiftApiClient.preloadIntrospectionResources(introspectionData),
```

Comment:
To be consistent might just want to name this as a normal query since all queries "preload" normalized resources?